File: backend/src/main/java/com/inventoryiq/domain/model/ProductStatus.java
package com.inventoryiq.domain.model;

/**
 * Sección 4.12 — Máquina de estados de un producto en una sucursal.
 * El orden de los valores no implica prioridad de evaluación; esa lógica
 * vive en ProductStatusEvaluator.
 */
public enum ProductStatus {
	NORMAL,
	REQUIRES_REPLENISHMENT,
	CRITICAL,
	OVERSTOCK,
	LOW_ROTATION
}

File: frontend/src/shared/productStatus.ts
import type { ProductStatus } from '../api/types'

/** Sección 4.12 de la documentación — máquina de estados de un producto. */
export const STATUS_LABELS: Record<ProductStatus, string> = {
  NORMAL: 'Normal',
  REQUIRES_REPLENISHMENT: 'Requiere reposición',
  CRITICAL: 'Crítico',
  OVERSTOCK: 'Sobrestock',
  LOW_ROTATION: 'Baja rotación',
}

export function statusClassName(status: ProductStatus): string {
  return `status status-${status.toLowerCase().replaceAll('_', '-')}`
}
